torch/export/exported_program.py

```diff
@@ -994,16 +1000,33 @@ def run_decompositions(
         `Core ATen Operator Set <https://pytorch.org/docs/stable/torch.compiler_ir.html>`_.
 
         For now, we do not decompose joint graphs.
+
+        If you want to disable a decomposition for certain operator, you can do following:
+
+        .. code-block:: python
+
+            ep = torch.export.export(model, ...)
+            ep = ep.run_decompositions(decomp_table={your_op: None})
+
+        If you want to get a core aten operator set except for certain operator, you can do following:
```

Comment:
cc @SherlockNoMad @avikchaudhuri This API unfortunately introduces a behaviour where {op: None} is a no-op for some operations and not others which is kind of leaking implementation details. For example, if you just do decomp_table={}, it will basically be equivalant to setting None to all post-autograd ops. Is this ok? Should we explain the difference more in detail in our public doc?

Reply:
Can you take a crack at explaining the current implementation in a docstring? Like, with different cases and example for each case? Otherwise I'm having trouble arriving at your conclusion for `{}`.

Of course it would be preferable to not leak implementation details. We have a choice of plumbing the decomp table through or interpreting / processing it how we want. So one possibility:

1. `decomp_table=None` is a special case, equivalent to default decompositions. 
2. `decomp_table={op1: decomp1, op2: decomp2}` should only decompose `op1` and `op2` and nothing else. No CIA either. If that means we have to internally extend the `decomp_table` given by the user to add None everywhere, so be it.
3. We should have an API returns a `decomp_table` equivalent to `decomp_table=None`, and it need not be just `core_aten_decompositions`. Is there a way we can generate what the decompositions for CIA would be here as well? Can the user alter these decompositions? If not, this proposal fails.
4. In this world, we don't need a user-facing `op: None` at all.

If (3) is not possible, then is there any other proposal that doesn't leak implementation details?

Reply:
According to @avikchaudhuri's proposal, it means our default decomp behaviour is at right before CIA decomp runs (similar to functional pre-dispatch IR). Then our op: None stuff will work well. @bdhirsh do you think we should expose option for user to attach custom decomp to CIA ops? In my naive understanding, it should be doable, although not possible before PTC.

Reply:
it's definitely doable: one of the original risks was that they swap out a CIA decomp with a "non-differentiable" decomp - but if this is all part of a "lower to inference IR" API that that point is moot anyway, since the user is telling us they won't be running autograd anyway (side note: do we raise an error if a user tries to take an inference IR graph and run it through autograd today?)

Reply:
I think we can incrementally add support for attaching custom decomp to CIA, and error out for now (without mentioning CIA, maybe just say that something like "this op doesn't support custom decomp right now."

Without that part, is there anything else that's hard in my proposal that cannot land before PTC?

Reply:
one question I have about the proposal is - how much do we want to avoid leaking CIA implementation details in the API?

(2) fixes part of the problem, but (1) (having the default be "decompose CIA ops") is probably still a bit confusing to users. So if our goal is to make CompositeImplicitAutograd decomps fully an implementation detail of the tracing process and not exposed to users, we would probably want the default to be "don't decompose anything". To be fair, if we get (2) from your proposal working, I don't think tweaking the default to be "decompose nothing" is much extra work.

@tugsbayasgalan can comment on the PTC deadline: I imagine that getting a PR up that changes the behavior before PTC is technically doable, although there's some tail risk that there are some failures-at-a-distance that we need to debug more carefully.

Reply:
Default can be anything though (assigning a meaning to `None`, which we pick to be the most likely thing you want). You don't decompose anything with `{}` in the proposal.